verilog/formatting/tree_unwrapper.cc

```diff
@@ -817,9 +850,15 @@ void TreeUnwrapper::Visit(const verible::SyntaxTreeLeaf& leaf) {
     StartNewUnwrappedLine();
   }
 
-  // Advances NextUnfilteredToken(), and extends CurrentUnwrappedLine().
-  // Should be equivalent to AdvanceNextUnfilteredToken().
-  AddTokenToCurrentUnwrappedLine();
+  if (tag == ',' && CurrentUnwrappedLine().IsEmpty()) {
+    // Partition would begin with a comma,
+    // instead add this token to previous partition
+    AppendTokenToPreviousUnwrappedLine();
```

Comment:
I *think* this is equivalent to adding tokens to the current partition followed by merging the last two partitions, like here:
https://github.com/google/verible/blob/81d39b1e13688850fb918cef9b5c47f135a962dd/common/formatting/token_partition_tree.h#L78

However, I'm a little concerned that if a `/* comment */` appears before `,`, that tokens might get added to the wrong partition, in the wrong order.  Can we test that out?  It's possible I didn't handle this correctly with `;` elsewhere.

It may be safer as a policy only add to the current partition, and perform adjustments (like moving token span boundaries) after the partitions are finalized.

Reply:
I changed the code to just use `MergeLastTwoPartitions`, I'm not sure why that wasn't working for me before.
I tested that on some examples with comments and it looks like it is moved correctly. I added some test cases to cover that.

> It may be safer as a policy only add to the current partition, and perform adjustments (like moving token span boundaries) after the partitions are finalized.

It is now done similarly to the `;` handling in https://github.com/google/verible/blob/master/verilog/formatting/tree_unwrapper.cc#L776. Since the commas can occur in the middle of a node, we would need to traverse the partition tree again looking for such lines. I think it could be done later, maybe in [MakeUnwrappedLinesWorklist](https://github.com/google/verible/blob/master/verilog/formatting/formatter.cc#L102)?